tests/file_checks/test_views.py

```diff
@@ -0,0 +1,233 @@
+            '{"failure": {"error": "File did not pass the virus scan", "status_code": 400}}',
+        ),
+    ),
+)
+def test_virus_check_puts_value_in_cache(
+    client,
+    mocker,
+    antivirus,
+    antivirus_scan_result,
+    expected_first_cache_value,
+    expected_second_cache_value,
+):
+    antivirus.scan.return_value = antivirus_scan_result
+    mock_redis_set = mocker.patch("app.redis_client.set")
+
+    _file_checks(client, b"%PDF-1.4 first file contents")
+    _file_checks(client, b"second file contents")
+
+    assert mock_redis_set.call_args_list == [
+        call(
+            "file-checks-b923c205dab97514f00194b3ee5cde0546f1aa7c",
+            expected_first_cache_value,
+            ex=86_400,
+        ),
+        call(
+            "file-checks-9e1c0c215d2eaefe915dd2d431a1cf21e777bbae",
+            expected_second_cache_value,
+            ex=86_400,
+        ),
+    ]
+
+
+def test_virus_check_returns_value_from_cache(client, mocker):
+    mock_redis_get = mocker.patch(
+        "app.redis_client.get",
+        return_value='{"failure": {"error": "I’m a teapot", "status_code": 418}}'.encode(),
+    )
+
+    file_1_content = b"%PDF-1.4 first file contents"
+    file_2_content = b"%PDF-1.4 second file contents"
+
+    for _ in range(3):
+        response_1 = _file_checks(client, file_1_content)
+        response_2 = _file_checks(client, file_2_content)
+
+        assert response_1.status_code == response_2.status_code == 418
+        assert response_1.json == response_2.json == {"error": "I’m a teapot"}
+
+    assert mock_redis_get.call_args_list == [
+        call("file-checks-b923c205dab97514f00194b3ee5cde0546f1aa7c"),
+        call("file-checks-fcdd443163779531f4fc93f72b34504ad6d14ac8"),
+        call("file-checks-b923c205dab97514f00194b3ee5cde0546f1aa7c"),
+        call("file-checks-fcdd443163779531f4fc93f72b34504ad6d14ac8"),
+        call("file-checks-b923c205dab97514f00194b3ee5cde0546f1aa7c"),
+        call("file-checks-fcdd443163779531f4fc93f72b34504ad6d14ac8"),
+    ]
+
+
+def test_different_cache_keys_for_different_filename_and_is_csv(client, mocker):
+    mock_redis_get = mocker.patch(
+        "app.redis_client.get",
+        return_value='{"failure": {"error": "I’m a teapot", "status_code": 418}}'.encode(),
+    )
+
+    file_content = b"%PDF-1.4 file contents"
+
+    _file_checks(client, file_content, service_id=UUID(int=0, version=4))
+    _file_checks(client, file_content, service_id=UUID(int=1, version=4))
+    _file_checks(client, file_content, service_id=UUID(int=1, version=4))
+
+    assert mock_redis_get.call_args_list == [
+        call("file-checks-01ca5a1f80d254b3d6f4dcff59c1d93b4f2ec939"),
+        call("file-checks-dc50228dab1b172ca18465eef02df493d208896b"),
+        call("file-checks-dc50228dab1b172ca18465eef02df493d208896b"),
+    ]
+
+
+def test_different_cache_keys_for_different_service_ids(client, mocker):
```

Comment:
This test seems to be actually testing for different csv. There seems to be a mistake in naming the tests as this test and the one above just need the names swapped around.